wallet/createtx.go

```diff
@@ -181,6 +183,13 @@ func (w *Wallet) txToOutputs(outputs []*wire.TxOut,
 			)
 
 			for _, e := range eligible {
+				// Restrict the selected utxos if a filter
+				// function is provided.
+				if allowUtxo != nil &&
```

Comment:
nowhere in the API docs do we state that a nil predicate has "always return true" semantics. I am not sure how and where to communicate this, but I may suggest that we either add it to the api docs or do away with it entirely and just pass in

```go
always := func(_ A) bool { return true }
```

Reply:
ahh you mean keeping as is but just add `always` in tests where we would prior use `nil` ?

I think we need go 1.21 for this at least thats what my vs-code linter says:

Even if I do it like this:

```
// always defines a generic filter function which always returns true.
func always[A any](A) bool {
	return true
}
```

and supply it to the function, it tells me:

```
implicitly instantiated function as argument requires go1.21 or latercompiler[UnsupportedFeature](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#UnsupportedFeature)
```

So I just added documentation for now.

Reply:
Yeah the monomorphic one is fine by me. Also if you really think that leaving nil in place is preferred I can be convinced, but I do think that it means that we need to be consistent about the semantics of `nil` predicates (functions from some A to bool). If we do this, we should commit to a convention that `nil` predicates are equivalent to the `always` I defined above.

Reply:
I like the design choice to only using `nil` where we conventionally defined it otherwise it's not good practice. So for sure will stay with the `always` function approach.